plugin/core/sessions.py

```diff
@@ -279,7 +279,12 @@ def _handle_request_workspace_configuration(self, params: Dict[str, Any], reques
         items = []  # type: List[Any]
         requested_items = params.get("items") or []
         for requested_item in requested_items:
-            items.append(self.config.settings)  # ???
+            if requested_item['section'] in self.config.settings:
+                items.append(self.config.settings[requested_item['section']])
+            elif requested_item['section'] == '':  # required for ESLint server
```

Comment:
IMO this should just check if the value is false'y so `elif not requested_item['section']`.
The code is supposed to follow the spec, not any particular server, and per spec the section can be null/undefined also.
Also no need for the comment. Same reason, this code is supposed to follow spec, not be server-specific.

Reply:
And the falsey'y check could be done first. Not a big deal in practice though.

Reply:
Also, since the `section` key can be missing in `requested_item`, there should be an extra `in` check to avoid crashing in that case.

Reply:
What should be returned if the `section` key is missing? I'm thinking the entire settings dict would be a suitable return value, or `null`.

Reply:
I think we can keep the current behavior of returning client's `settings` object. This allows for handling the simplest of cases at least, where people can define values to return up-front in the config. But ultimately, it's not very well defined what should be returned and some cases (eslint's workspacefolder) can't be handled with simple settings and thus might need server-specific handling in LanguageHandler.

Reply:
That workspacefolder settings from eslint can probably be solved by allowing template strings like `${folder}` and `${project_path}` etc in the settings.

I agree returning the entire settings dict seems suitable if the section key is missing.

Reply:
Thanks, should be fixed now in https://github.com/sublimelsp/LSP/pull/908/commits/96a0e3959fce3a83071184d1ac2d3a8902fb473c